src/assets/assets.service.ts

```diff
@@ -34,18 +35,30 @@ export class AssetsService {
     private readonly transactionsService: TransactionsService
   ) {}
 
-  public async findOne(ticker: string): Promise<Asset> {
-    return await this.polymeshService.polymeshApi.assets.getAsset({ ticker }).catch(error => {
+  public async findOne(asset: string): Promise<Asset> {
+    let getAssetPromise;
+    if (asset.length === ASSET_ID_LENGTH) {
```

Comment:
There should be a function `isAssetId` that captures the actual logic.

```ts
const assetIdRegex = /0x[0-9A-F]{32}/;
const isAssetId = (id: string): boolean => {
  return assetIdRegex.test(id);
};
```

Then the code can be shared with the validator and we always perform the most exact check we can.